site/zenodo_rdm/support/support.py

```diff
@@ -0,0 +1,120 @@
+from invenio_accounts.sessions import _extract_info_from_useragent
+from marshmallow import ValidationError
+from werkzeug.utils import cached_property
+
+from .email import SupportEmailService
+from .errors import ConfirmationEmailNotSent, SupportEmailNotSent
+from .schema import SupportFormSchema
+
+
+class ZenodoSupport(MethodView):
+    """Zenodo support view."""
+
+    def __init__(self):
+        self.template = "zenodo_rdm/support.html"
+        self.email_service = SupportEmailService()
+        self.support_form_schema = SupportFormSchema()
+
+    def get(self):
+        """Renders the support template."""
+        user_agent = _extract_info_from_useragent(request.headers.get("User-Agent"))
+        browser_client = user_agent.get("browser", "")
+        browser_version = user_agent.get("browser_version", "")
+        browser_string = browser_client + " " + browser_version
+        platform = user_agent.get("os", "")
+        system_info = {"browser": browser_string, "platform": platform}
+
+        return render_template(
+            self.template,
+            categories=self.categories,
+            system_info=system_info,
+        )
+
+    def post(self):
+        """Receives a form, validates its data and handles it."""
+        data = {**request.form.to_dict(), "files": request.files.getlist("files")}
+
+        try:
+            data = self.validate_form(data)
+        except (ValidationError) as e:
+            return str(e), 400
+
+        try:
+            self.handle_form(data)
+        except (Exception) as e:
+            return str(e), 500
+
+        return redirect(url_for("invenio_app_rdm.index"))
+
+    def handle_form(self, form_data):
+        """Form controller."""
+        self.send_support_email(form_data)
+        self.send_confirmation_email(form_data)
+
+    def validate_form(self, form_data):
+        """Validates form using a schema"""
+        return self.support_form_schema.load(form_data)
+
+    def send_support_email(self, data):
+        """Send an email to support"""
+        try:
+            self.email_service.send_support_email(
+                sender_name=data.get("name"),
+                email=data.get("email"),
+                description=data.get("description"),
+                send_user_agent=data.get("sysInfo"),
+                subject=data.get("subject"),
+                category=data.get("category"),
+                files=data.get("files"),
+                title="[{}]: {}".format(data.get('category'), data.get('subject'))
+            )
+        except (smtplib.SMTPSenderRefused) as e:
+            raise SupportEmailNotSent(
+                "There was an issue sending an email to the provided "
+                "address (ours), please make sure it is correct. "
+                "If this issue persists you can send "
+                "us an email directly to {}".format(self.email_service.support_emails)
+            )
+        except (Exception) as e:
```

Comment:
this will catch any exception and will make debugging harder, is it on purpose ?

Reply:
it emulates the original one https://github.com/zenodo/zenodo/blob/master/zenodo/modules/support/views.py#L68, with more granular error catching (e.g. not sending the support email is not the same as not sending the confirmation email). 

 I agree that we lose the message and it will a nightmare to debug, especially since it is working with e-mail. Is there a standard way of logging exceptions in invenio?